cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/ExecutionSparkContext.java

```diff
@@ -160,7 +195,8 @@ public ExecutionSparkContext(SparkSpecification specification, Id.Program progra
 
   @Override
   public synchronized <T extends Dataset> T getDataset(String name, Map<String, String> arguments) {
-    Map<String, String> datasetArgs = RuntimeArguments.extractScope(Scope.DATASET, name, getRuntimeArguments());
+    Map<String, String> datasetArgs = new HashMap<>();
+    datasetArgs.putAll(RuntimeArguments.extractScope(Scope.DATASET, name, getRuntimeArguments()));
     datasetArgs.putAll(arguments);
```

Comment:
No objection to do a defensive copy here. I also think it's good to make the contract for the extractScope clear.